File: Java/src/main/java/QuestionBank/Solution401_500/Solution485.java
package QuestionBank.Solution401_500;

/**
 * @author windows
 */
public class Solution485 {
    public int findMaxConsecutiveOnes(int[] nums) {
        int count = 0;
        int max = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1){
                if (i > 0 && nums[i - 1] == 1) {
                    count++;
                }else{
                    count = 1;
                }
            }else{
                count = 0;
            }
            max = Math.max(max, count);
        }
        return max;
    }


    public static void main(String[] args) {
        Solution485 solution485 = new Solution485();
    }
}

File: C++/src/Solution401_500/Solution485.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution485.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution485::findMaxConsecutiveOnes(vector<int> &nums) {
    int max = 0;
    int count = 0;
    for (int i = 0; i < nums.size(); ++i) {
        if (nums[i] == 1){
            if (++count > max){
                max = count;
            }
        }else{
            count = 0;
        }
    }
    return max;
}
